Highway.cpp

```diff
@@ -13,34 +16,43 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
+    // assert(false);
 
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
+    
+    if (auto* car = dynamic_cast<Car*>(v)) { car->closeWindows(); }
+    if (auto* motorcycle = dynamic_cast<Motorcycle*>(v)) { motorcycle->lanesplitAndRace(); }
+    
```

Comment:
- [ ] use `else-if`.  Once the cast succeeds, stop trying to cast to the other types.
- [ ] you forgot about your `SemiTrucks`